pkg/controller/apmserver/apmserver_controller.go

```diff
@@ -176,34 +176,19 @@ var _ driver.Interface = &ReconcileApmServer{}
 // Reconcile reads that state of the cluster for a ApmServer object and makes changes based on the state read
 // and what is in the ApmServer.Spec
 func (r *ReconcileApmServer) Reconcile(request reconcile.Request) (reconcile.Result, error) {
-	// atomically update the iteration to support concurrent runs.
-	currentIteration := atomic.AddInt64(&r.iteration, 1)
-	iterationStartTime := time.Now()
-	log.Info("Start reconcile iteration", "iteration", currentIteration, "namespace", request.Namespace, "as_name", request.Name)
-	defer func() {
-		log.Info("End reconcile iteration", "iteration", currentIteration, "took", time.Since(iterationStartTime), "namespace", request.Namespace, "as_name", request.Name)
-	}()
-
-	// Fetch the ApmServer resource
-	as := &apmv1alpha1.ApmServer{}
-	err := r.Get(request.NamespacedName, as)
+	defer common.LogReconciliationRun(log, request, &r.iteration)()
```

Comment:
Since I had to modify the reconciliation loops of almost all the operators anyway, I decided to clean them up a little bit and in the process included this tiny change which has bugged me for quite a while. In hindsight, I probably should have done it in a different PR as you say :)